subsys/net/lib/tls_credentials/tls_credentials.c

```diff
@@ -74,6 +75,37 @@ struct tls_credential *credential_next_get(sec_tag_t tag,
 	return NULL;
 }
 
+sec_tag_t credential_next_tag_get(sec_tag_t iter)
+{
+	int i;
+	sec_tag_t lowest = TLS_SEC_TAG_NONE;
+
+	/* Scan all slots and find lowest sectag greater than iter */
+	for (i = 0; i < ARRAY_SIZE(credentials); i++) {
+		/* Skip empty slots. */
+		if (credentials[i].type == TLS_CREDENTIAL_NONE) {
+			continue;
+		}
+
+		/* Skip any slots containing sectags not greater than iter */
+		if (credentials[i].tag <= iter && iter != TLS_SEC_TAG_NONE) {
+			continue;
+		}
+
+		/* Find the lowest of such slots */
+		if (lowest == TLS_SEC_TAG_NONE || credentials[i].tag < lowest) {
+			lowest = credentials[i].tag;
```

Comment:
Shouldn't you just `break` here?

Reply:
Scanning the whole array is required since the array is not guaranteed to be sorted by sectag -- there may be even lower sectags later in the array.

The phrasing of the comment is a little vague, I will see if I can make it clearer